charmcraft/package.py

```diff
@@ -89,7 +89,17 @@ def launch_shell(*, cwd: Optional[pathlib.Path] = None) -> None:
 class Builder:
     """The package builder."""
 
-    def __init__(self, *, config, force, debug, shell, shell_after, measure):
+    def __init__(
+        self,
+        *,
+        config,
+        force,
+        debug,
+        shell,
+        shell_after,
+        measure,
+        use_shared_cache=True,
```

Comment:
Let's rather push the shared cache path into the builder on creation and make it the creator's task to determine the cache path.